test/unittest/unittest_base_properties.cpp

```diff
@@ -96,6 +96,19 @@ TEST(BasicProperty, tagCast) {
   }
 }
 
+TEST(BasicProperty, propInfo) {
+  { /**< prop_info test */
+    using prop_type = nntrainer::prop_info<QualityOfBanana>::prop_type;
```

Comment:
Let's say differently. for the property input_shape.
then
1. prop_type is input_shape.
2. tag_type is dimension
3. data_type is string 

??

Reply:
Yes, that is current behavior 👍. For `#2`, this is left to incorporate with different property parser. (eg, dimension type for ini but maybe `string` type for tflite.)